src/index.js

```diff
@@ -336,7 +336,8 @@ function removeNodeFromHost(host, node) {
 }
 
 function removeSlotFromRoot(root, node) {
-  node.assignedNodes().forEach(slotNodeFromSlot);
+  const assignedNodes = Array.prototype.slice.call(node.assignedNodes());
```

Comment:
I might just leave it then. `Array.prototype.forEach.call(node.assignedNodes(), slotNodeFromSlot);` doesn't work unfortunately, it actually needs to clone the nodes to avoid the problem where it modifies the array it is looping over